quam_builder/architecture/quantum_dots/README.md

```diff
@@ -0,0 +1,682 @@
+
+#### System Control
+
+- `ramp_to_zero(ramp_duration: Optional[int] = None)`  
+  Ramps all channels to zero and resets integrated voltage tracking. The `ramp_duration` parameter can be a QUA variable.
+
+- `apply_compensation_pulse(max_voltage: float = 0.49)`  
+  Applies compensation pulses to counteract integrated voltage drift (when tracking enabled).
+
+### 8.3 Combining Physical and Virtual Gates
+
+You can seamlessly combine physical and virtual gates in the same operation. The system automatically resolves all contributions and applies them additively to the physical channels.
+
+## 9. Underlying Architecture and Voltage Resolution
+
+### 9.1 How Virtual Gate Resolution Works
+
+When you apply a voltage operation on virtual gates, the system automatically converts this to voltages applied to the underlying physical gates using matrix transformations. Here's how it works:
+
+1. **Matrix-Based Transformation**: Each `VirtualisationLayer` defines a transformation matrix that maps virtual gate voltages to target gate voltages (either physical gates or lower-level virtual gates).
+
+2. **Multi-Layer Resolution**: For multi-layer virtualisation, the system processes layers from the outermost (highest-level virtual gates) to the innermost (physical gates), applying the inverse of each transformation matrix.
+
+3. **Additive Contributions**: All contributions from virtual gates at different layers are summed together at the physical gate level, allowing for complex control schemes.
+
+### 9.2 Core Allocation and Performance
+
+**One core is dedicated to each physical gate**, regardless of the number of virtual gates or virtualisation layers. This has important implications:
+
+- **Scalable Performance**: Adding virtual gates or virtualisation layers doesn't increase the computational load on the QUA system
+- **Real-Time Operation**: All matrix calculations are performed at compile time, not during execution
+- **Predictable Resource Usage**: The number of cores required is determined solely by the number of physical channels
+
+For example, if you have 3 physical gates (`P1`, `P2`, `P3`) but 10 virtual gates across 3 virtualisation layers, you still only need 3 cores - one for each physical gate.
+
+### 9.3 Matrix Calculation Example
+
+Consider a simple two-layer system:
+
+```python
+# Layer 1: Virtual gates to physical gates
+matrix_1 = [[1.0, 0.5], [0.5, 1.0]]  # v_Coarse1, v_Coarse2 -> P1, P2
+
+# Layer 2: Higher-level virtual gates to Layer 1 virtual gates
+matrix_2 = [[0.1, 0.0], [0.0, 0.1]]  # v_FineTune1, v_FineTune2 -> v_Coarse1, v_Coarse2
+```
+
+When you set `v_FineTune1 = 0.1V`, the system:
+
+1. Applies inverse of matrix_2: `v_Coarse1 = 0.1V / 0.1 = 1.0V`
+2. Applies inverse of matrix_1: `P1 = 1.0V * 1.0 + 0V * 0.5 = 1.0V`, `P2 = 1.0V * 0.5 + 0V * 1.0 = 0.5V`
+
+This transformation happens at compile time, so the QUA program only sees the final physical gate voltages.
+
+## 10. Relevance to Spin Qubits
```

Comment:
I would actually put this to the top, to start with why we did this and provide some context